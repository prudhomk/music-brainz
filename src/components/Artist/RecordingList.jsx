import React, { useState, useEffect } from 'react';
import Recording from './Recording';
import { Link, useParams } from 'react-router-dom';
import PropTypes from 'prop-types';
import { mungeRecordings, getRecordings } from '../../services/musicApi';


export default function RecordingList() {
  const [recordings, setRecordings] = useState([]);
  const [loading, setLoading] = useState(true);
  const { id } = useParams();

  useEffect(async () => {
    const recordings = await getRecordings(id);
   
    const mungedRecordings = await mungeRecordings(recordings);
    console.log(mungedRecordings);
    await setRecordings(mungedRecordings);
    
    setLoading(false);
  }, []);

  if(recordings) {
    const recordingResult = recordings.map((recording) => (
      <>
        <li key={recording.id}>
          <Link to={`/${recording.id}/lyrics`}>
            <Recording {...recording} />
          </Link>
        </li>
      </>
    
    ));

    return (
      <>
        <h1>Recording List</h1>
        <ul>
          {recordingResult}
        </ul>
      </>
    );
  }


  return (
    <div>
      placeholder
    </div>
  );
}

RecordingList.propTypes = {
  recordingList: PropTypes.arrayOf(
    PropTypes.shape({
      title: PropTypes.string.isRequired,
    })
  )
};

