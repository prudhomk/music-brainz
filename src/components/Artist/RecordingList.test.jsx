import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { render, screen, waitFor } from '@testing-library/react';
import { rest } from 'msw';
import { setupServer } from 'msw/node'; 
import RecordingList from './RecordingList';

const server = setupServer(
  rest.get(
    'http://musicbrainz.org/ws/2/release?artist=<ARTIST_ID>&fmt=json',
    (req, res, ctx) => {
      return res(
        ctx.json(
          {
            'id': '1',
            'title': 'banana'
          }
        )
      );
    }
  )
);

describe('Main display component snapshot', () => {
  beforeAll(() => server.listen());
  afterAll(() => server.close());

  it('render RecordingList', async () => {
    const { asFragment } = render(
      <MemoryRouter>
        <RecordingList />
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
