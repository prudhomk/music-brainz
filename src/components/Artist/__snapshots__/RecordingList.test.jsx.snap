// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main display component snapshot render RecordingList 1`] = `
<DocumentFragment>
  <h1>
    Recording List
  </h1>
  <ul />
</DocumentFragment>
`;
